f3dex2.s

```diff
@@ -55,6 +55,15 @@ overlay_imem equ 0x0006
 // RSP DMEM
 .create DATA_FILE, 0x0000
 
+/*
+Matrices are stored like in OpenGL, where adjacent elements form a column
+vector. In C, this would be matrix[col][row]. For the integer part:
+00 08 10 18  typical  Xscl Rot  Rot  Xpos
```

Comment:
I have mixed feelings on documenting matrices in this orientation. I think row-major makes more sense to use, as it lines up with how the bytes are ordered in memory (i.e. translation on the bottom row instead of the right-most column). This would also line up with most other documentation that can be found about the N64 and the F3D family microcodes.

Reply:
If you feel strongly that they should be switched, I can do that. I'm not familiar with much in terms of documentation beyond the official SGI docs, and I haven't seen anything that explains the matrix format. If you could link to the documentation you're referring to, that would be great. I think most people will know 3D graphics math with things the other way around.

Reply:
The SGI docs you mentioned have an example of the matrix format where they place the translation on the bottom row: 
![image](https://user-images.githubusercontent.com/68165316/199755267-e740d388-f9a8-4fed-9405-e4d21964a56e.png)
The OoT and SM64 decomps also use that notation since it's needed to make an array declared in C line up with the F3D element ordering. An example is here: https://github.com/zeldaret/oot/blob/ab9030006220d756753b3d8eb914777872c586b2/src/code/z_skin_matrix.c#L5. While the matrix in that example is a purely diagonal one, if you wanted to adjust translation for example you'd do so by modifying the elements on the bottom row rather than the rightmost column.

Reply:
Sorry for the delay. I was looking at the SGI RSP documentation and the SGI general system manual but not the SGI F3D*** manual. I have changed all the documentation to have the matrix laid out this way.